File: Frame_APT/app/src/main/java/com/ct/apt/MainActivity.java
package com.ct.apt;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.ct.apt_annotations.ARouter;
import com.ct.apt_annotations.Event;

/**
 * Android APT
 */


@ARouter(path = "/app/MainActivity")
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: DesignPatternsAndAlgorithms/app/src/main/java/com/ct/design/MainActivity.kt
package com.ct.design

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ViewGroup

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
}

class A(context: Context) : ViewGroup(context) {
    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
        getChildMeasureSpec(0,0,0)
    }

}